include/mbedtls/pk.h

```diff
@@ -170,6 +176,31 @@ typedef size_t (*mbedtls_pk_rsa_alt_key_len_func)( void *ctx );
 #endif /* MBEDTLS_PK_RSA_ALT_SUPPORT */
 
 /**
+ * \brief           Check whether instances of one PK-type are convertible
+ *                  to those of another.
+ *
+ * \param from      PK-type to convert from
+ * \param to        PK-type to convert to
+ *
+ * \note            Convertibility is currently restricted to the following two cases:
+ *                  (1) The target PK-type has the same underlying
+ *                      mathematical entity but is restricted in its
+ *                      functionality. For example, the type PK_ECKEY
+ *                      of unrestricted EC keys converts to PK_ECDSA
+ *                      of EC keys to use for ECDSA signing only, but
+ *                      not conversely.
+ *                  (2) The target PK-type has the same underlying
+ *                      mathematical entity but uses it differently
+ *                      for some of its functionality. For example,
+ *                      PK_RSA converts to PK_RSASSA_PSS which has uses
+ *                      different padding scheme for signatures.
+ *
+ * \return          0 if \c from is convertible to \c to, nonzero otherwise.
+ *
+ */
+int mbedtls_pk_type_convertible( mbedtls_pk_type_t from, mbedtls_pk_type_t to );
```

Comment:
I'm slightly worried about the name and return convention. IMO it would be clearer to either: rename it as "check_convertible" as we have a consistent convention that "check_foo" returns 0 if foo is true, or rename as "type_is_convertible" and reverse the return value. Sorry for being picky, this kind of things has been an issue in the past.